src/components/common/FileUploader.tsx

```diff
@@ -0,0 +1,65 @@
+'use client';
+
+import * as React from 'react';
+
+import ClipLoader from 'react-spinners/ClipLoader';
+
+interface PropTypes {
+  onUpload: (formData: FormData) => void;
+  isLoading: boolean;
+}
+
+const FileUploader = (props: PropTypes) => {
+  const { onUpload, isLoading } = props;
+  const [file, setFile] = React.useState<File>();
+
+  const onFileAdd = (event: React.FormEvent<HTMLInputElement>) => {
+    if (event.currentTarget.files?.length) {
+      const currentFile = event.currentTarget.files[0];
+
+      setFile(currentFile);
+    }
+  };
+
+  const onSubmit = async () => {
+    if (!file) return;
+    const formData = new FormData();
+    formData.append('keywords', file);
+    formData.append('userId', '1');
+
+    await onUpload(formData);
+    setFile(undefined);
+  };
+
+  return (
+    <div className="max-w-lg rounded overflow-hidden shadow-lg w-full p-4">
+      {(!isLoading && (
+        <div className="mb-4">
+          <h3 className="mb-4">Upload the csv file with keywords you want to search</h3>
+          <input
+            type="file"
+            data-testid="file-uploader"
+            name="keywords"
+            onChange={onFileAdd}
```

Comment:
It is better to keep the prop value with its name:

```suggestion
            onChange={onChange}
```